src/nvim/eval.c

```diff
@@ -8018,26 +8019,36 @@ static void f_bufnr(typval_T *argvars, typval_T *rettv)
     rettv->vval.v_number = -1;
 }
 
-/*
- * "bufwinnr(nr)" function
- */
-static void f_bufwinnr(typval_T *argvars, typval_T *rettv)
+static void buf_win_common(typval_T *argvars, typval_T *rettv, bool get_nr)
 {
-  (void)get_tv_number(&argvars[0]);         /* issue errmsg if type error */
-  ++emsg_off;
+  (void)get_tv_number(&argvars[0]);  // issue errmsg if type error
```

Comment:
`bufname()` describes that `{expr}` is a `Number` or `String`. Why not just check for these types (and if they don't match, give an error message and return). Otherwise we have this:
`:echo bufwinid(1.2) `        gives an error message
`:echo bufwinid(v:true)`     gives no error message